### ISSUE TITLE
Strange behavior with using .reduce on Combinatorics.combination

### ISSUE THREAD

**Issue by SwiftWinds** [dankogai/js-combinatorics#62]
For example, if I were to have:
```js
const Combinatorics = require("js-combinatorics");

const group = ["Penelope", "Brad", "Ivan"];

const cmb = Combinatorics.combination(group, 2);

const pairings = cmb.reduce((acc, pair) => `${acc}${pair[0]} ${pair[1]}, `, "");

console.log(pairings);
```
The output would be `P e, B r, I v, ` instead of the expected `Penelope Brad, Penelope Ivan, Brad Ivan, `.

Is there any specific reason why this happens besides that it's just a bug? Other array methods like `forEach`, `filter`, `map`, etc. seem to work fine; just not `reduce`.

Thanks! 😊

**Comment by SwiftWinds** [comment]
A simple example of this bug is in this repo: https://github.com/SwiftWinds/js-combinatorics-bug-with-reduce

**Comment by SwiftWinds** [comment]
As a side note, a current workaround would be to have:
```js
const cmb = Combinatorics.combination(group, 2).toArray();
```

**Comment by dankogai** [comment]
Hmm.  It is not exactly a bug since `Combinatorics` *does not* define `.reduce()`.  The problem is that combinatorics implements lazy evaluation by redefining `Array` functions.  But  I did not mask out `reduce` so `reduce()` still exists, which iterates through the hidden, opaque storage of `Combinatorics` object, which is generatated via `Object.create(ary.slice())`. 

Anyway it is confusion so I will add `.reduce()`.

**Comment by SwiftWinds** [comment]
Ok, thanks! :)

**Comment by dankogai** [comment]
Now that I have released version 0.5.5 with `.reduce()` , I'm closing this issue now.

**Comment by SwiftWinds** [comment]
Awesome.
